File: ndb/05_binary_search/example01/main.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    public static void main(String[] args) {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        try {
            // n 입력 받기
            System.out.println("생성할 원소 개수, 찾을 문자열 입력. ex)3 Kim");
            String[] input = br.readLine().split(" ");
            int n = Integer.parseInt(input[0]);
            String target = input[1];

            System.out.println("원소 개수만큼 문자열 입력. ex)Lee Kim Park");
            BufferedReader br2 = new BufferedReader(new InputStreamReader(System.in));
            String[] array = br2.readLine().split(" ");

            System.out.println(sequentialSearch(n, target, array));
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }

    // 순차 탐색
    public static int sequentialSearch(int n, String target, String[] array) {
        // 각 원소 하나씩 확인
        for (int i = 0; i < n; i++) {
            if (array[i].equals(target)) {
                return i + 1;
            }
        }
        return 0;
    }
}

File: ndb/05_binary_search/example01/main.dart
import 'dart:io';

void main(List<String> arguments) {
  print("생성할 원소 개수, 찾을 문자열 입력. ex)3 Kim");
  var input1 = stdin.readLineSync()?.split(' ');
  if (input1 != null) {
    // n, target 입력 받기
    var n = int.parse(input1[0]);
    var target = input1[1];

    print("원소 개수만큼 문자열 입력. ex)Lee Kim Park");
    var array = stdin.readLineSync()?.split(' ');
    if (array != null) {
      print(sequentialSearch(n, target, array));
    }
  }
}

int sequentialSearch(int n, String target, List<String> array) {
  // 각 원소 하나씩 확인
  for (int i = 0; i < n; i++) {
    if (array[i] == target) {
      return i + 1;
    }
  }
  return 0;
}
